### ISSUE TITLE
Invariant failed: You should not use <Switch> outside a <Router>

### ISSUE THREAD

**Issue by davidnguyen179** [storybookjs/storybook#12823]
**Describe the bug**

I am using `storybook` & `react-router`, but when I integrate them together, it throws exception.

**Screenshots**
<img width="872" alt="Screen Shot 2020-10-20 at 9 36 04" src="https://user-images.githubusercontent.com/6290720/96526125-b6325880-12b7-11eb-9aad-28fae54a88dc.png">

**Additional context**

Version of packages I use

- react-router: 5.1.2
- react-router-dom: 5.1.2
- @storybook/react: 5.0.11

**Comment by shilman** [comment]
Are you wrapping your stories with a `Router` using a decorator?

https://storybook.js.org/docs/react/writing-stories/decorators#context-for-mocking

**Comment by davidnguyen179** [comment]
Thanks I was able to make it work. So I wrapped around my component with `MemoryRouter` or `BrowserRouter`. But after I did that I was not able to navigate using `react router` (`history.push()`, or `Link` tag) in `storybook`. So I have to switch using `MemoryRouter` (that keeps the history of your “URL” in memory and it does not read or write to the address bar) to make my component can navigate using `react-router`
